blog/2024-05-20-standing-up-a-fhir-gateway-for-dhis2-from-an-ig/index.md

```diff
@@ -0,0 +1,553 @@
+---
+slug: 2024/05/standing-up-a-fhir-gateway-for-dhis2-from-an-ig
+title: Standing up a FHIR Gateway for DHIS2 from an Implementation Guide
+authors: claudemamo
+tags: [fhir, camel, ig, java]
+---
+
+A modern vision of health interoperability is one where systems exchange health care data using the [FHIR (Fast Health Interoperability Resources)](https://en.wikipedia.org/wiki/Fast_Healthcare_Interoperability_Resources) standard to describe the types of resources and data to be exchanged. FHIR outlines these resources and data using what is called an [Implementation Guide (IG)](https://www.hl7.org/fhir/implementationguide.html). An IG specifies the rules for interoperability in a formal, machine-readable, and testable way. In this post, we will show you how to stand up a FHIR gateway for DHIS2 from an IG so that DHIS2 can interoperate with FHIR clients.
+
+<!--truncate-->
+
+The following diagram illustrates the role of a FHIR gateway, or more accurately, a FHIR facade sitting in front of DHIS2:
+
+![FHIR gateway](./fhir-gateway.png)
+
+A FHIR facade is a layer of software translating FHIR exchanges between a FHIR client and a non-FHIR system. [FHIR consists of numerous communication patterns](https://build.fhir.org/exchange-module.html) therefore this how-to will focus only on one: the [RESTful API](https://www.hl7.org/fhir/http.html). More precisely, we will demonstrate step-by-step how to build an _IG-first_ FHIR facade that returns a FHIR [QuestionnaireResponse resource](https://build.fhir.org/questionnaireresponse.html) from DHIS2 given a [tracked entity](https://docs.dhis2.org/en/develop/using-the-api/dhis-core-version-240/tracker.html#tracked-entity) ID. IG-first is emphasised because this approach to building facades takes advantage of IG computability and all the benefits that go with it (e.g., enforcing separation of concerns between the IG author and the gateway implementer).
+
+The facade in this how-to will (1) use the transmitted QuestionnaireResponse ID, as sent by the FHIR client, to fetch a tracked entity from DHIS2 to then (2) convert this entity into a QuestionnaireResponse thanks to the [IG StructureMap](https://build.fhir.org/structuremap.html) before (3) returning it to the FHIR client. The engine powering the facade is [Apache Camel](https://camel.apache.org/manual/faq/what-is-camel.html): a Java integration framework for building message-oriented middleware.
+
+Before drilling in, it is worth highlighting that there is no special reason why QuestionnaireResponse was selected to be the FHIR representation of a tracked entity other than it is conceptually easier to map from a tracked entity to a QuestionnaireResponse. Your data exchange requirements will dictate which FHIR resources to use for representing DHIS2 resources.
```

Comment:
I think it should be `QuestionnaireResponse` as opposed to QuestionnaireResponse? (so single quotes for an inline codeblock?